### PR TITLE
Add Clearfog-Pro Support

### PR DESCRIPTION
**Pull request by Valks**
Adding clearfog-pro support. Main change over base is the 6 port switch is enabled.

Also removed the Mender patches as they should be in the meta-mender-clearfog(-pro) layers instead.

I also have a patch for mender-community which adds clearfog-pro support based on this repo.

### CODE REVIEW

File: recipes-kernel/linux/linux-stable-common.inc [hunk 1]
```diff
@@ -3,7 +3,6 @@ require recipes-kernel/linux/linux-yocto.inc
 
 SRC_URI = "\
     https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-${PV}.tar.xz \
-    file://defconfig \
```
Comment (mirzak) [inline recipes-kernel/linux/linux-stable-common.inc]: This is no need for creating a separate `linux-stable` recipes for the different boards as the only difference is the `defconfig` file.

The directory structure should be enough, e.g:

```
linux-stable/clearfog/defconfig
linux-stable/clearfog-pro/defconfig
``` 

The current `linux-stable_4.19.bb` recipe would pick up the correct file depending on the machine name.

So leave this recipe as is.

File: recipes-kernel/linux/linux-stable-common.inc [hunk 2]
```diff
@@ -22,8 +21,4 @@ SRCREV = "6a31767f84ad31445865f1297d49937319f775c3"
 
 PV = "${LINUX_VERSION}"
 
-# Override COMPATIBLE_MACHINE to include your machine in a copy of this recipe
-# file. Leaving it empty here ensures an early explicit build failure.
-COMPATIBLE_MACHINE = "clearfog"
```
Comment (mirzak) [inline recipes-kernel/linux/linux-stable-common.inc]: Update to,

```
COMPATIBLE_MACHINE = "(^clearfog$|^clearfog-pro$)"
```

File: recipes-kernel/linux/linux-stable_clearfog-base_4.19.bb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require linux-stable-common.inc
+
+SRC_URI_append = " file://clearfog/defconfig"
+
+# Override COMPATIBLE_MACHINE to include your machine in a copy of this recipe
+# file. Leaving it empty here ensures an early explicit build failure.
+COMPATIBLE_MACHINE = "clearfog"
```
Comment (mirzak) [inline recipes-kernel/linux/linux-stable_clearfog-base_4.19.bb]: See comment above regarding this file

File: recipes-kernel/linux/linux-stable_clearfog-pro_4.19.bb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require linux-stable-common.inc
+
+SRC_URI_append = " file://clearfog-pro/defconfig"
+
+# Override COMPATIBLE_MACHINE to include your machine in a copy of this recipe
+# file. Leaving it empty here ensures an early explicit build failure.
+COMPATIBLE_MACHINE = "clearfog-pro"
```
Comment (mirzak) [inline recipes-kernel/linux/linux-stable_clearfog-pro_4.19.bb]: See comment above regarding this file

File: recipes-bsp/u-boot/patches/0001-configs-clearfog-Mender-integration.patch [hunk 5]
```diff
@@ -1,37 +0,0 @@
...
--#endif
--#endif
-+#define CONFIG_BOOTCOUNT_LIMIT
-+#define CONFIG_BOOTCOUNT_ENV
- 
- #if defined(CONFIG_CMD_SCSI) || defined(CONFIG_CMD_BUBT) || defined(CONFIG_ENV_IS_IN_SATA)
- #define CONFIG_SYS_SATA_ENV_DEV		0
--- 
-2.11.0
-
```
Comment (mirzak) [inline recipes-bsp/u-boot/patches/0001-configs-clearfog-Mender-integration.patch]: Nice catch. Agree that these should be deleted but please create a separate commit for this.

**Review by mirzak — CHANGES_REQUESTED** [review]
Nice work.

Dropped some comments and change requests.

Also please sign-off your commits, per,

https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines

**Comment by Valks** [comment]
OK, all changes have been done and commits signed. Hopefully now it's ready.

File: recipes-bsp/u-boot/patches/0001-configs-clearfog-Mender-integration.patch (hunk 5 above)
Comment (Valks) [reply recipes-bsp/u-boot/patches/0001-configs-clearfog-Mender-integration.patch]: Please see: https://github.com/mirzak/meta-clearfog/pull/4

File: recipes-bsp/u-boot/patches/0001-configs-clearfog-Mender-integration.patch [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+-#endif
+-#endif
++#define CONFIG_BOOTCOUNT_LIMIT
++#define CONFIG_BOOTCOUNT_ENV
+ 
+ #if defined(CONFIG_CMD_SCSI) || defined(CONFIG_CMD_BUBT) || defined(CONFIG_ENV_IS_IN_SATA)
+ #define CONFIG_SYS_SATA_ENV_DEV		0
+-- 
+2.11.0
+
```
Comment (mirzak) [inline recipes-bsp/u-boot/patches/0001-configs-clearfog-Mender-integration.patch]: You seem to be adding the Mender files here, which where removed in #4

**Review by mirzak — CHANGES_REQUESTED** [review]
Sorry for the delay, been on the road.

Can you please squash/fixup the first 2 commits?

File: recipes-bsp/u-boot/patches/0001-configs-clearfog-Mender-integration.patch (hunk 6 above)
Comment (Valks) [reply recipes-bsp/u-boot/patches/0001-configs-clearfog-Mender-integration.patch]: Yes, my intention was for this patch to be applied first. Hopefully it's suitable now.

File: recipes-kernel/linux/linux-stable_clearfog_4.19.bb [hunk 7]
```diff
@@ -22,8 +22,8 @@ SRCREV = "6a31767f84ad31445865f1297d49937319f775c3"
```
Comment (mirzak) [inline recipes-kernel/linux/linux-stable_clearfog_4.19.bb]: This rename is not necessary,

```
recipes-kernel/linux/linux-stable_4.19.bb → ...ernel/linux/linux-stable_clearfog_4.19.bb
```

**Review by mirzak — CHANGES_REQUESTED** [review]
One more thing then this one is good to go.

File: recipes-kernel/linux/linux-stable_clearfog_4.19.bb (hunk 7 above)
Comment (Valks) [reply recipes-kernel/linux/linux-stable_clearfog_4.19.bb]: Reverted back.

**Review by mirzak — APPROVED** [review]

**Comment by mirzak** [comment]
Thank you for working on this. Merging.
